src/main/java/seedu/address/logic/parser/AddDoctorCommandParser.java

```diff
@@ -0,0 +1,52 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NRIC;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.AddDoctorCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Nric;
+import seedu.address.model.person.doctor.Doctor;
+
+/**
+ * Parses input arguments and creates a new AddDoctorCommand object
+ */
+public class AddDoctorCommandParser implements NewParser<AddDoctorCommand> {
```

Comment:
Utilise "AddCommandParserTest.java" to help you write test for all the methods here